ui/components/component-library/checkbox/checkbox.scss

```diff
@@ -1,10 +1,11 @@
 .mm-checkbox {
+  cursor: pointer;
```

Comment:
Allows for the cursor to be a pointer on the label


https://github.com/MetaMask/metamask-extension/assets/8112138/6b09655e-53e4-401b-aed2-dab08987f6d1